src/Select2/Select2.tsx

```diff
@@ -0,0 +1,209 @@
+  NO_OPTIONS_FOUND: "Select2--notFound",
+};
+
+/*
+  Basic single-item-select drop down component
+*/
+const Select2: React.FC<Props> = ({
+  className,
+  name,
+  label,
+  hideLabel,
+  options,
+  clearable,
+  onChange,
+  size,
+}) => {
+  const [selectableOptions, setSelectableOptions] = useState(options);
+  const {
+    isOpen,
+    getToggleButtonProps,
+    getLabelProps,
+    getMenuProps,
+    getInputProps,
+    getComboboxProps,
+    highlightedIndex,
+    getItemProps,
+    openMenu,
+    selectedItem,
+    selectItem,
+  } = useCombobox<Option>({
+    items: selectableOptions,
+    itemToString: (o) => (o ? o.value : ""),
+    onStateChange: (changes) => {
+      const { type } = changes;
+      switch (type) {
+        case useCombobox.stateChangeTypes.InputBlur:
+          // reset any text that has been entered
+          if (selectedItem) {
+            selectItem(selectedItem);
+            break;
+          }
+
+          selectItem(null);
+          break;
+        default:
+          break;
+      }
+    },
+    onIsOpenChange: (changes) => {
+      const { type } = changes;
+      switch (type) {
+        // if we're opening the drop down for the first time
+        // show everything even if something had already been selected previously
+        case useCombobox.stateChangeTypes.FunctionOpenMenu:
+          setSelectableOptions(options);
+          break;
+        default:
+          break;
+      }
+      return;
+    },
+    onInputValueChange: ({ inputValue }) => {
+      const inputLowerCase = inputValue.toLowerCase();
+      setSelectableOptions(options.filter((o) => o.value.toLowerCase().includes(inputLowerCase)));
+    },
+    onSelectedItemChange: (item) => {
+      if (onChange) {
+        onChange(item.selectedItem ? item.selectedItem.value : "");
+      }
+    },
+  });
+
+  const id = name;
+  const inputRef = useRef<HTMLInputElement>();
+  return (
+    <div className={classNames(cssClass.CONTAINER, formElementSizeClassName(size), className)}>
+      <label
+        className={classNames(cssClass.LABEL, hideLabel && cssClass.LABEL_HIDDEN)}
+        {...getLabelProps()}
+      >
+        {label}
+      </label>
+      <div
+        className={classNames(
+          cssClass.SELECT_CONTAINER,
+          isOpen && cssClass.SELECT_CONTAINER_FOCUSED,
+        )}
+        {...getComboboxProps({
+          onClick: (e) => {
+            if (!isOpen) {
+              openMenu();
+              if (inputRef.current) {
+                inputRef.current.select();
+              }
+            }
+          },
+        })}
+      >
+        <input
+          id={id}
+          name={id}
+          className={cssClass.INPUT}
+          {...getInputProps({
+            ref: inputRef,
+            onClick: (e) => {
+              if (!isOpen) {
+                openMenu();
+                if (inputRef.current) {
+                  inputRef.current.select();
+                }
+              }
+            },
```

Comment:
It looks like this callback function is the same as the onClick handler a few lines above, so you could reuse the same function if you wanted to save a couple of lines of code. It's not quite at the "rule of three" threshold, so feel free to ignore this.

Reply:
Turns out I don't think I need the one on the `<input>`. The parent click handler is enough 👍